libcontainer/cgroups/fs/apply_raw.go

```diff
@@ -301,6 +301,21 @@ func (m *Manager) Freeze(state configs.FreezerState) error {
 	return nil
 }
 
+// ThawAll unfreezes the container's freezer cgroup, and all subcgroups
+// recursively.
+func (m *Manager) ThawAll() error {
+	err := m.Freeze(configs.Thawed)
+	if err != nil {
+		return err
+	}
+	paths := m.GetPaths()
+	freezer, err := subsystemsLegacy.Get("freezer")
+	if err != nil {
+		return err
+	}
+	return freezer.(*FreezerGroup).RecursiveThaw(paths["freezer"])
```

Comment:
As now you have RecursiveThaw for cgroupv2, you can use
```go
freezer, err := m.getSubsystems().Get("freezer")
if err != nil {
    return err
}
return freezer.RecursiveThaw(paths["freezer"])
```

Reply:
Maybe I'm misunderstanding, but in order to call `RecursiveThaw` directly don't we need to add `RecursiveThaw` as a method on the `subsystem` interface?  Which in turn now every implementation of the `subsystem` interface will need to have a dummy `RecursiveThaw` method?  I agree the cast isn't nice but the alternative also doesn't seem very palpable...maybe I'm not following?

Reply:
Right, my mistake, we could have a RecursiveSet method for subsystem, but we can do that when we see more needs.